components/script/dom/headers.rs

```diff
@@ -217,6 +225,41 @@ impl Headers {
     pub fn extract_mime_type(&self) -> Vec<u8> {
         self.header_list.borrow().get_raw("content-type").map_or(vec![], |v| v[0].clone())
     }
+
+    pub fn sort_header_list(&self) -> Vec<(String, String)> {
+        let borrowed_header_list = self.header_list.borrow();
+        let headers_iter = borrowed_header_list.iter();
+        let mut header_vec = vec![];
+        for header in headers_iter {
+            let name = header.name().to_string();
+            let value = header.value_string();
+            let name_value = (name, value);
+            header_vec.push(name_value);
+        }
+        header_vec.sort();
+        header_vec
+    }
+}
+
+impl Iterable for Headers {
+    type Key = ByteString;
+    type Value = ByteString;
+
+    fn get_iterable_length(&self) -> u32 {
+        self.header_list.borrow().iter().count() as u32
+    }
+
+    fn get_value_at_index(&self, n: u32) -> ByteString {
+        let sorted_header_vec = self.sort_header_list();
+        let value = sorted_header_vec[n as usize].1.clone();
+        ByteString::new(value.into_bytes().to_vec())
+    }
+
+    fn get_key_at_index(&self, n: u32) -> ByteString {
+        let sorted_header_vec = self.sort_header_list();
+        let key = sorted_header_vec[n as usize].1.clone();
```

Comment:
I suspect this should be `.0` instead... do any test results change when that's corrected?